pkg/integrations/elasticsearch_exporter/elasticsearch_exporter.go

```diff
@@ -66,6 +70,21 @@ type Config struct {
 	ExportDataStreams bool `yaml:"data_stream,omitempty"`
 	// Export stats for Snapshot Lifecycle Management
 	ExportSLM bool `yaml:"slm,omitempty"`
+	// BasicAuth block allows secure connection with Elasticsearch cluster via Basic-Auth
+	BasicAuth *promCfg.BasicAuth `yaml:"basic_auth,omitempty"`
+}
+
+// Custom http.Transport struct for Basic Auth-secured communication with ES cluster
+type BasicAuthHTTPTransport struct {
```

Comment:
nit: can we rename this to `AuthenticatedHTTPTransport` or sth like this, since it just takes a generic header value?